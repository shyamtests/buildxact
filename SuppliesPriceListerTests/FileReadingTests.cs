using ApprovalTests;
using ApprovalTests.Reporters;
using buildxact_supplies.DataSources;
using Newtonsoft.Json;
using Xunit;

namespace SuppliesPriceListerTests
{
    [UseReporter(typeof(DiffReporter))]
    public class FileReadingTests
    {
        [Fact]
        public void HumphriesReadsEmbeddedFilesAndConvertsToBuildingSupplies()
        {
            var supplies = new HumphriesSupplies().GetBuildingSupplies();
            
            Approvals.VerifyJson(JsonConvert.SerializeObject(supplies));
        }
        
        [Fact]
        public void MegaCorpReadsEmbeddedFilesAndConvertsToBuildingSupplies()
        {
            var supplies = new MegacorpSupplies().GetBuildingSupplies();
            
            Approvals.VerifyJson(JsonConvert.SerializeObject(supplies));
        }
    }
}
