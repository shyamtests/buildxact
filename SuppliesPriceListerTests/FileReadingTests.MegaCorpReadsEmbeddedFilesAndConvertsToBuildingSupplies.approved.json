[
    {
        "Id": "1",
        "Name": "100 x 200 x 20mpa Internal Beam",
        "Price": 40.0
    },
    {
        "Id": "0",
        "Name": "100 x 350 Thickened Edge",
        "Price": 35.0
    },
    {
        "Id": "2",
        "Name": "1000 x 1000 x 350 x 20mpa Square Pad Footing",
        "Price": 12.25
    },
    {
        "Id": "3",
        "Name": "100mm x 25mpa Slab",
        "Price": 202.35
    },
    {
        "Id": "4",
        "Name": "100mm x 32 mpa Slab",
        "Price": 22.5
    },
    {
        "Id": "6",
        "Name": "100 x 200 x 20mpa Internal Beam",
        "Price": 2.0
    },
    {
        "Id": "7",
        "Name": "11 x 200 Trench Mesh",
        "Price": 14.96
    }
]